app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt

```diff
@@ -549,7 +549,12 @@ class ChatPresenter(chatActivity: ChatActivity) : IChatPresenter, IChatModel.OnR
         }
 
         if (c > 0) {
-            chatView?.askForPermission(permissionsGranted)
+            val permissionToGet = arrayOfNulls<String>(c)
+            for(i in 0..(c-1)){
+                permissionToGet[i] = permissionsGranted[i]
+            }
```

Comment:
Where are you removing nulls?

Reply:
@iamareebjamal 
If the nulls are passed to the askForPermissions method then a NullPointerException is caused in the onRequestPermissionResult method which causes the crash.

Reply:
So, Where are you removing nulls?

Reply:
I am copying the non-null values (first 'c' values) of the permissionsGranted to a new array permissionToGet and then passing permissionsToGet to the function.

Reply:
I'm not seeing any check where you only copy non-null values

Reply:
The check is done by the value of the variable 'c'. In the array permissionsGranted the first c values are initialised and the new array is of size=c only

Reply:
I realised what I described in the last comment can not be seen in the 'changes'. 
Apologies for that. It was my first pull request and I'll try to be more careful  and descriptive next time.